### PR TITLE
Allow creating custom templates in user-config directory

### PR DESCRIPTION
**Pull request by hweissi**
This commit makes the `template` commandline tool try to get template files out of `~/.config/pwntools/templates` before using the ones shipped with the library.

This would people to fix issues like #2030 permanently on their machine, across updates.
Could also be used to import frequently used libraries automatically, or strip down the template to a more minimal one.

### CODE REVIEW

File: pwnlib/commandline/template.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 def main(args):
     lookup = TemplateLookup(
-        directories      = [os.path.join(pwnlib.data.path, 'templates')],
+        directories      = [os.path.expanduser('.config/pwntools/templates/'), os.path.join(pwnlib.data.path, 'templates')],
```
Comment (peace-maker) [inline pwnlib/commandline/template.py]: The path should start with a `~` to expand to the home directory.

**Review by peace-maker — CHANGES_REQUESTED** [review]
This should be documented in the command help string as well.

**Comment by peace-maker** [comment]
Maybe adding a `--template` argument to override it per invocation is the nicest way to add this. Then we can mention the .config folder in the help string of that option.

**Comment by hweissi** [comment]
I added a short help string, but it does not explain where to get the base template from. It isn't too hard to find, but still requires looking into the pwntools repo. I'm not sure where to document that though, seems a bit too much for an argument help string

**Comment by peace-maker** [comment]
Cool, but I think it's weird to pass a directory where the template has to be named `pwnup.mako` as argument. I'd expect it to point to the template file itself. Can you change it to expect a path to a file and use basename to extract the template filename please?

For the .config folder we need to document the filename too.

**Comment by peace-maker** [comment]
We could lookup the install location and add the full local path to the data/templates directory to the help? I'm not sure where else to document this either since it's very specific to the `pwn template` command.

**Comment by hweissi** [comment]
I tried a version that falls back to the base template if the chosen template is invalid - it lead to pretty unclean code though. Letting it crash when the template path is invalid is probably cleaner, and also more clear to the user.

I also added code that displays the full path to the template in the argument description, but it's probably going to look weird in the documentation. Is there a way to see if the docs are being built in the code?

**Comment by hweissi** [comment]
Bump

File: pwnlib/commandline/template.py [hunk 2]
```diff
@@ -22,10 +22,16 @@
 parser.add_argument('--path', help='Remote path of file on SSH server')
 parser.add_argument('--quiet', help='Less verbose template comments', action='store_true')
 parser.add_argument('--color', help='Print the output in color', choices=['never', 'always', 'auto'], default='auto')
+parser.add_argument('--base', help='Path to a custom base template. Defaults to \'~/.config/pwntools/templates/pwnup.mako\'. '
```
Comment (peace-maker) [inline pwnlib/commandline/template.py]: I think `--template` matches the matter better than `--base`. This isn't a base template, which is extended by other templates, but THE template used to generate the script. Yes `pwn template --template` looks weird, but I can't think of a better name.

The mention of the `.config` template folder should say something like "if it exists"? Right now it sounds like the default template is in the `~/.config` folder while saying the other default template is in yet another path :D

**Review by peace-maker — CHANGES_REQUESTED** [review]
Sorry for the delay! [Stack overflow](https://stackoverflow.com/questions/20843737/check-if-sphinx-doc-called-the-script) says you can check if you're building the docs using `if 'sphinx' in sys.modules:`. Maybe that works to say just `pwnlib/data/templates` when building docs?

**Comment by hweissi** [comment]
> Sorry for the delay! [Stack overflow](https://stackoverflow.com/questions/20843737/check-if-sphinx-doc-called-the-script) says you can check if you're building the docs using `if 'sphinx' in sys.modules:`. Maybe that works to say just `pwnlib/data/templates` when building docs?

That solution seems a bit hacky to me, but it works - I added it to the PR, as well as the other requested changes

**Review by peace-maker — APPROVED** [review]
Looks good, thank you!

**Pull request merged**